nox/tasks.py

```diff
@@ -180,10 +185,11 @@ def filter_manifest(
             return 3
 
     # Filter by python interpreter versions.
-    # This function never errors, but may cause an empty list of sessions
-    # (which is an error condition later).
     if global_config.pythons:
         manifest.filter_by_python_interpreter(global_config.pythons)
+        if not manifest:
```

Comment:
This is a subtle change in behavior. Combining `--list` with `--python` or `--keywords` should not lead to an error, even if no sessions are selected. In other words, `--list` should always produce a listing.

```suggestion
        if not manifest and not global_config.list_sessions:
```